pontoon/messaging/views.py

```diff
@@ -1,13 +1,93 @@
 import json
+import uuid
+
+from guardian.decorators import permission_required_or_403
+from notifications.signals import notify
 
 from django.contrib.auth.decorators import login_required
+from django.core.exceptions import PermissionDenied
+from django.core.mail import EmailMultiAlternatives
 from django.db import transaction
 from django.http import JsonResponse
 from django.shortcuts import get_object_or_404, render
 from django.utils import timezone
 from django.views.decorators.http import require_POST
 
 from pontoon.base.models import UserProfile
+from pontoon.base.utils import require_AJAX
+from pontoon.messaging import forms, utils
+
+
+def messaging(request):
+    if not request.user.has_perm("base.can_manage_project"):
+        raise PermissionDenied
+
+    return render(
+        request,
+        "messaging/messaging.html",
+    )
+
+
+@permission_required_or_403("base.can_manage_project")
+@require_AJAX
+@require_POST
+@transaction.atomic
+def send_message(request):
+    # Send notifications
+    if request.method == "POST":
+        form = forms.MessageForm(request.POST)
+
+        if not form.is_valid():
+            return JsonResponse(dict(form.errors.items()))
+
+        # TODO: implement recipient filters
+        from django.contrib.auth.models import User
+
+        recipients = User.objects.filter(pk=request.user.pk)
+
+        is_notification = form.cleaned_data.get("notification")
+        is_email = form.cleaned_data.get("email")
+        subject = form.cleaned_data.get("subject")
+        body = form.cleaned_data.get("body")
+
+        if is_notification:
+            identifier = uuid.uuid4().hex
```

Comment:
Is this actually used for anything?

Reply:
Yes, that allows us to find all instances of the same notification (one instance is created for each user as per `django-notifications-hq`).

To implement #3246, we'll need a new data model and than we can drop this.